src/consensus/vote.cpp

```diff
@@ -549,6 +564,106 @@ std::string VoteManager::getJsonStr(std::vector<Vote> const& votes) {
   return ptroot.toStyledString();
 }
 
+std::vector<Vote> VoteManager::getProposalVotes(uint64_t const pbft_round) {
+  std::unordered_map<blk_hash_t, std::unordered_set<vote_hash_t>> proposal_voted_value_map;
+  {
+    sharedLock_ lock(verified_votes_access_);
+    proposal_voted_value_map = verified_votes_[pbft_round][1];
+  }
+
+  std::vector<Vote> proposal_votes;
+  for (auto const& voted_value : proposal_voted_value_map) {
+    for (auto const& v_hash : voted_value.second) {
+      auto vote = db_->getVerifiedVote(v_hash);
+      if (vote) {
+        proposal_votes.emplace_back(*vote);
+      } else {
+        LOG(log_er_) << "Cannot get vote " << v_hash << " from DB";
+      }
+    }
+  }
+
+  return proposal_votes;
+}
+
+votesBundle VoteManager::getVotesBundleByRoundAndStep(uint64_t const round, size_t const step,
+                                                      size_t const two_t_plus_one) {
+  blk_hash_t voted_block_hash;
+  std::vector<vote_hash_t> votes_hash;
+
+  {
+    sharedLock_ lock(verified_votes_access_);
+    if (verified_votes_.count(round)) {
+      if (verified_votes_[round].count(step)) {
+        for (auto const& voted_value : verified_votes_[round][step]) {
+          if (voted_value.second.size() >= two_t_plus_one) {
+            voted_block_hash = voted_value.first;
+            votes_hash.reserve(two_t_plus_one);
```

Comment:
I guess this is some mistake. Why would you reserve for the same size over and over again inside for loop ?

Reply:
Because only need 2t+1 number of votes to make consensus